src/App.tsx

```diff
@@ -15,36 +15,87 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Alphabetically = 'alphabetically',
+  Length = 'length',
+  None = 'none',
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setReversed] = useState<boolean>(false);
+  const [sortType, setSortType] = useState<SortType>(SortType.None);
+
+  const reverseGoods = () => {
+    setReversed(!isReversed);
```

Comment:
```suggestion
    setReversed(isReversed => !isReversed);
```

If you change state based on previous value you should always use callback